src/main/java/net/fabricmc/loom/util/ModProcessor.java

```diff
@@ -58,74 +61,68 @@
-		}
-
-		remapJar(input, output, project, artifact);
+	public static void processMod(Project project, File input, Function<String, File> outputFileProducer, Iterable<File> dependencies) throws IOException {
+		final File output = outputFileProducer.apply(input.getName());
+	    remapJar(project, input, output, dependencies);
 
 		//Enable this if you want your nested jars to be extracted, this will extract **all** jars
 		if (project.getExtensions().getByType(LoomGradleExtension.class).extractJars) {
-			handleNestedJars(input, project, config, artifact);
+			handleNestedJars(project, input, outputFileProducer);
 		}
 
 		//Always strip the nested jars
 		stripNestedJars(output);
 	}
 
-	private static void handleNestedJars(File input, Project project, Configuration config, ResolvedArtifact artifact) throws IOException {
-		JarFile jarFile = new JarFile(input);
-		JarEntry modJsonEntry = jarFile.getJarEntry("fabric.mod.json");
-
-		if (modJsonEntry == null) {
-			return;
-		}
-
-		try (InputStream inputStream = jarFile.getInputStream(modJsonEntry)) {
-			JsonObject json = GSON.fromJson(new InputStreamReader(inputStream), JsonObject.class);
-
-			if (json == null || !json.has("jars")) {
-				return;
-			}
-
-			JsonArray jsonArray = json.getAsJsonArray("jars");
+    private static void handleNestedJars(
+                    Project project,
+                    File input,
+                    Function<String, File> outputFileProducer) throws IOException {
+        JarFile jarFile = new JarFile(input);
+        JarEntry modJsonEntry = jarFile.getJarEntry("fabric.mod.json");
 
-			for (int i = 0; i < jsonArray.size(); i++) {
-				JsonObject jsonObject = jsonArray.get(i).getAsJsonObject();
-				String fileName = jsonObject.get("file").getAsString();
-				project.getLogger().lifecycle(String.format("Found %s nested in %s", fileName, input.getName()));
-				processNestedJar(jarFile, fileName, project, config, artifact);
-			}
-		}
-	}
+        if (modJsonEntry == null) {
+            return;
+        }
 
-	private static void processNestedJar(JarFile parentJar, String fileName, Project project, Configuration config, ResolvedArtifact artifact) throws IOException {
-		LoomGradleExtension extension = project.getExtensions().getByType(LoomGradleExtension.class);
+        try (InputStream inputStream = jarFile.getInputStream(modJsonEntry)) {
+            JsonObject json = GSON.fromJson(new InputStreamReader(inputStream), JsonObject.class);
 
-		JarEntry entry = parentJar.getJarEntry(fileName);
+            if (json == null || !json.has("jars")) {
+                return;
+            }
 
-		if (entry == null) {
-			throw new RuntimeException(String.format("%s was not found in %s", fileName, parentJar.getName()));
-		}
+            JsonArray jsonArray = json.getAsJsonArray("jars");
 
-		File nestedFile = new File(extension.getNestedModCache(), fileName.substring(fileName.lastIndexOf("/")));
+            for (int i = 0; i < jsonArray.size(); i++) {
```

Comment:
can't you use a `for : ` (foreach) loop here?

Reply:
It would iterate over `JsonElement`s rather than `JsonObject`s so it's hardly much better

Reply:
It would have the same benefits as in any other for loop, 
``` JsonObject jsonObject = jsonElement.getAsJsonObject();```

Reply:
TBH This is original code, i did not change this. But I plan to go over all of this and clean this kind of stuff up in an upcoming PR later on. This will probably stay in this PR, getting this to behave is already kind of difficult.